### PR TITLE
chore: only run subquery rewrite for sql contains subquery

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: src/query/sql/src/planner/optimizer/s_expr.rs [hunk 1]
```diff
@@ -172,4 +167,56 @@ impl SExpr {
     pub(crate) fn applied_rule(&self, rule_id: &RuleID) -> bool {
         self.applied_rules.get(rule_id)
     }
+
+    /// Check if contain subquery
+    pub(crate) fn contain_subquery(&self) -> bool {
+        if !find_subquery(&self.plan) {
+            for child in self.children.iter() {
+                return child.contain_subquery();
```
Comment (sundy-li) [inline src/query/sql/src/planner/optimizer/s_expr.rs:175]: early return inside a loop.

**Comment by xudong963** [comment]
@mergify update

File: src/query/sql/src/planner/optimizer/s_expr.rs [hunk 2]
```diff
@@ -171,9 +171,7 @@ impl SExpr {
     /// Check if contain subquery
     pub(crate) fn contain_subquery(&self) -> bool {
         if !find_subquery(&self.plan) {
-            for child in self.children.iter() {
-                return child.contain_subquery();
-            }
+            return self.children.iter().any(|child| child.contain_subquery());
```
Comment (sundy-li) [inline src/query/sql/src/planner/optimizer/s_expr.rs:174]: should check if children are empty, it will return true.

Comment (sundy-li) [reply src/query/sql/src/planner/optimizer/s_expr.rs:174]: !self.children.is_empty() && ...

Comment (xudong963) [reply src/query/sql/src/planner/optimizer/s_expr.rs:174]: An empty iterator returns false

image

**Pull request merged**